### DISCUSSION TITLE
Can BookPlayer Read Audiobook files already on the device that are currently used by Apple Books

### DISCUSSION THREAD

**Discussion by LabLover** [TortugaPower/BookPlayer#1011] [Q&A] [answered]
Can BookPlayer Read Audiobook files already on the device that are currently used by Apple Books?  Or Do I have to upload those to the cloud first?  Also, can I use file sharing in iTunes?  Should I zip each folder and share the zip file with BookPlayer?  Will Bookplayer automatically uncompress each zip file?  Thanks in advance, these are probably novice questions, which is what I am with Bookplayer.

**Comment by GianniCarlo** [comment, marked as answer]
Hi @LabLover ! directly from Apple Books no, we currently don't have the integration for it yet. We do support iTunes file sharing, you can just drop the zip file, and BookPlayer will unzip it and prompt you if you want to move them into a (new) folder or volume. 

Regarding importing from different cloud providers, we have an integration with the native Files app, so basically you can import from any cloud provider app that exposes their files with the Files app (all the big ones do), and can select multiple files at a time for import. The only difference I've found between different cloud provider apps, is the convenience they provides, for example if you'd rather share directly from the OneDrive app to BookPlayer (and not go through the Files app interface), that app only lets you share one item at a time, because when multiple items are selected, it generates a link instead of sharing the actual selected items (I think Google Drive does share the individual items, but I may be misremembering things)

The other option for importing is Airdrop, which on iOS 17 has become a tad inconvenient, as when the drop finishes, it no longer shows a list of available apps to open the files, instead, it imports directly to the Files app, and leaves a notification so you can tap it and go to the shared file(s), and from there you can 'share' again, and open it in BookPlayer from the share sheet

**Reply by LabLover** [reply]
Hi GianniCarlo!  Thank you so much for your excellent explanation, it was the perfect reply!  Also, thank you very much for an excellent evolving app which really fills a need in iOS for a high quality audiobook player.

**Reply by TheByteSmasher** [reply]
Hi! Is integration with the on device Apple Books database on the horizon or is it impossible? I use iTunes (windows) to manage and sync my books for ease of use, whereas file sharing is more manual and time consuming. If possible, it would be great to use iTunes to manage (add books and determine playlists to sync) my library but have BookPlayer read the library and play from there. Even if there was an extra step (on device file import/copy from books) to prevent "corruption" of books database it would be a great feature. I realize this would effectively double the library on my device but having this option would be great. Thanks.
